src/parser.sid

```diff
@@ -205,6 +263,32 @@
 		<err-expected-type>;
 	};
 
+        directive-properties: (dialect: re-dialect) -> () [
+                prop-schema: (dialect: re-dialect) -> (ps: ast-prop-schema) = {
+                        s = STRING;
+                        colon;
+                        sch = <parse-schema>;
+                        re = <new-regexp>(dialect,s);
+                        ps = <new-prop-schema>(re,sch);
+                };
+
+                list-of-properties: (dialect: re-dialect) -> (ps: ast-prop-schema) = {
+                        ps = prop-schema(dialect);
+                        {
+                                comma;
+                                ps1 = list-of-properties(dialect);
+                                <append-list-of-props>(ps,ps1);
+                        ||
+                                $;
+                        };
+                };
+        ] = {
+                OCURLY;
+                ps = list-of-properties(dialect);
+                <set-properties>(ps);
+                CCURLY;
+        };
+
```

Comment:
I've been figuring out how I feel about this. I understand the instinct to centralise the grammar here, and pass in a dialect. I don't generally like passing that sort of thing through SID's productions because they can't control the grammar - imagine if we had a different token for regexps than we did for strings here, for instance. But we don't, so I figure I'm okay with this.

I'm also very cautious about introducing types in SID, and `:re-dialect`, `:ast-prop-schema` both make me suspicious. But I don't have any particular objection, if we must have them.

Reply:
We could eliminate :re-dialect, but we'd end up duplicating the rules in several places.

:ast-prop-schema represents a (:regexp, :schema) tuple.  Since we're building a list of them, that seems fine to me, but I'm not attached to it.  My changes to the parser are mostly cobbled together from reading the sid docs, looking at your stuff, and trial-and-error. Do you have alternate ways to do this?

Reply:
None that I want to do. Like I said above, I'm okay with this.